test/MathUtils.test.cpp

```diff
@@ -0,0 +1,237 @@
+#include "NAS2D/MathUtils.h"
+#include "NAS2D/Renderer/Primitives.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+TEST(MathUtils, divideUp)
+{
+	EXPECT_DEATH(NAS2D::divideUp(0, 0), "divideUp(0,0) divide by zero.");
+	EXPECT_EQ(0, NAS2D::divideUp(0, 1));
+	EXPECT_EQ(0, NAS2D::divideUp(0, 2));
+	EXPECT_EQ(0, NAS2D::divideUp(0, 3));
+	EXPECT_DEATH(NAS2D::divideUp(1, 0), "divideUp(1,0) divide by zero.");
+	EXPECT_EQ(1, NAS2D::divideUp(1, 1));
+	EXPECT_EQ(1, NAS2D::divideUp(1, 2));
+	EXPECT_EQ(1, NAS2D::divideUp(1, 3));
+	EXPECT_DEATH(NAS2D::divideUp(2, 0), "divideUp(2,0) divide by zero.");
+	EXPECT_EQ(2, NAS2D::divideUp(2, 1));
+	EXPECT_EQ(1, NAS2D::divideUp(2, 2));
+	EXPECT_EQ(1, NAS2D::divideUp(2, 3));
+	EXPECT_EQ(1, NAS2D::divideUp(5, 8));
+	EXPECT_DEATH(NAS2D::divideUp(256, 0), "divideUp(256,0) divide by zero.");
+	EXPECT_EQ(256, NAS2D::divideUp(256, 1));
+	EXPECT_EQ(128, NAS2D::divideUp(256, 2));
+	EXPECT_EQ(86, NAS2D::divideUp(256, 3));
+	EXPECT_EQ(64, NAS2D::divideUp(256, 4));
+	EXPECT_EQ(52, NAS2D::divideUp(256, 5));
+	EXPECT_EQ(44, NAS2D::divideUp(256, 6));
+	EXPECT_EQ(38, NAS2D::divideUp(256, 7));
+	EXPECT_EQ(32, NAS2D::divideUp(256, 8));
+	EXPECT_EQ(29, NAS2D::divideUp(256, 9));
+	EXPECT_EQ(26, NAS2D::divideUp(256, 10));
+	EXPECT_EQ(24, NAS2D::divideUp(256, 11));
+	EXPECT_EQ(22, NAS2D::divideUp(256, 12));
+	EXPECT_EQ(20, NAS2D::divideUp(256, 13));
+}
+
+TEST(MathUtils, isPointInRectComponent)
+{
+	//Neghbors
+	EXPECT_EQ(false, NAS2D::isPointInRect(4, 4, 5, 5, 1, 1));
```

Comment:
I think this test really illustrates how hard the current method is to read at the point of call. There are 6 numbers, but it's not clear which ones represent the point and which ones represent the rect.

To be clear, this isn't a problem with the test case, but rather the test case highlighting a weakness of the design. It would be nice to update the method parameters to make for easier to read test code, which generally serves as good example code for how things will actually be used.